### PR TITLE
Add info about unsyncable docs

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Jira

https://jira.mongodb.org/browse/DOCSP-13396

### Staged Changes (Requires MongoDB Corp SSO)

https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/13396/mongodb/#realm-sync-schema-data-consistency

### CODE REVIEW

**Comment by tkaye407** [comment]
Looks great. I pinged mike to make sure we are ok with documenting the collection name where we store these. I think this is helpful information, but we also haven't been sharing the internal metadata information willy-nilly so just want to verify he is ok with this and then will LGTM 

Quick nits: 

1. Can we link to this section from this "Best Practices" section:https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/13396/sync/#best-practices 
2. Can we link to it / update the language here: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/13396/sync/errors/#mongodb-translator-errors

Also, meta-question, we dont seem to actually reference the translator much / if at all, is this intentional?

File: source/mongodb.txt [hunk 1]
```diff
@@ -191,6 +191,59 @@ might resemble the following code:
       { $unset: { _id_realm_transaction: "" } }
    )
 
+Realm Sync Schema Data Consistency
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the collection in which you create and modify objects is also accessed by Realm 
+:ref:`Sync clients <sync>`, you must ensure that the data conforms to the
```
Comment (nathan-contino-mongo) [inline source/mongodb.txt:198]:
```suggestion
If the collection in which you create and modify objects is also accessed by
:ref:`{+sync+} clients <sync>`, you must ensure that the data conforms to the 
```

File: source/mongodb.txt [hunk 2]
```diff
@@ -191,6 +191,59 @@ might resemble the following code:
       { $unset: { _id_realm_transaction: "" } }
    )
 
+Realm Sync Schema Data Consistency
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the collection in which you create and modify objects is also accessed by Realm 
+:ref:`Sync clients <sync>`, you must ensure that the data conforms to the 
+Realm :ref:`object schema <object-schema>` defined on the collection. Consider
```
Comment (nathan-contino-mongo) [inline source/mongodb.txt:199]:
```suggestion
{+service-short+} :ref:`object schema <object-schema>` defined on the collection. Consider 
```

File: source/mongodb.txt [hunk 3]
```diff
@@ -191,6 +191,59 @@ might resemble the following code:
...
 
+Realm Sync Schema Data Consistency
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the collection in which you create and modify objects is also accessed by Realm 
+:ref:`Sync clients <sync>`, you must ensure that the data conforms to the 
+Realm :ref:`object schema <object-schema>` defined on the collection. Consider 
+the following scenario:
+
+- You have a collection of objects that is used by {+realm+} Sync clients. The
```
Comment (nathan-contino-mongo) [inline source/mongodb.txt:202]:
```suggestion
- You have a collection of objects that is used by {+sync+} clients. The 
```
realm replacement is for lowercase-r realm database instances

File: source/mongodb.txt [hunk 4]
```diff
@@ -191,6 +191,59 @@ might resemble the following code:
...
+     :copyable: false
+
+     type Person = {
+       _id: BSON.ObjectId;
+       name: string;
+       address: string;
+       age: number;
+     };
+  
+- Using a Realm client (say, {+compass+}), you create and insert a document
```
Comment (nathan-contino-mongo) [inline source/mongodb.txt:215]:
```suggestion
- Using a {+service-short+} client (say, {+compass+}), you create and insert a document 
```

File: source/mongodb.txt [hunk 5]
```diff
@@ -191,6 +191,59 @@ might resemble the following code:
...
+synced, the Realm app is paused.
+
+To fix this problem, you must **replace** the invalid document with one that 
+conforms to the object model. Updating the existing document will not add it to 
+Sync. 
+
+.. note:: Summary
+
+   If your data is used by Sync clients but can also be created or 
+   modified outside of Realm Sync, you must ensure those creations and modifications
```
Comment (nathan-contino-mongo) [inline source/mongodb.txt:243]:
```suggestion
   modified outside of {+sync+}, you must ensure those creations and modifications 
```

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM with a few nits

File: source/mongodb.txt [hunk 6]
```diff
@@ -191,6 +191,59 @@ might resemble the following code:
...
+- The collection is now divergent: Sync clients do not see the new document, but 
+  {+service+} clients do. 
+
+The same scenario holds true if a valid document is modified outside of Sync 
+such that one or more properties no longer conform to the Sync schema. 
+
+Any document that fails the object schema validation is added to the 
+``__realm_sync.unsynced_documents`` collection (along with the reason for the 
+failure) and will also be logged in the 
+Realm UI ``Logs`` console. If there are 10,000 or more documents that can't be
```
Comment (tkaye407) [inline source/mongodb.txt:233]: Turns out this number is actually 100,000 :)

File: source/mongodb.txt [hunk 7]
```diff
@@ -191,6 +191,59 @@ might resemble the following code:
...
+  pushed to Sync.
+
+- The collection is now divergent: Sync clients do not see the new document, but 
+  {+service+} clients do. 
+
+The same scenario holds true if a valid document is modified outside of Sync 
+such that one or more properties no longer conform to the Sync schema. 
+
+Any document that fails the object schema validation is added to the 
+``__realm_sync.unsynced_documents`` collection (along with the reason for the
```
Comment (tkaye407) [inline source/mongodb.txt:231]: Can we add a note or something to make sure people realize it is ok if they read from this collection / db but they should never write to it?

File: source/mongodb.txt (hunk 6 above)
Comment (MongoCaleb) [reply source/mongodb.txt:233]: eh, what's an extra digit amongst friends?

**Review by tkaye407 — APPROVED** [review]
LGTM. Thanks for these changes!

**Pull request merged**